### ISSUE TITLE
Apple Mach-O Linker failure

### ISSUE THREAD

**Issue by DennisSnijder** [airbnb/react-native-maps#766]
When installing the React Native map module after running the pod install, and following the steps i get the following error while building in xcode

ld: framework not found React
clang: error: linker command failed with exit code 1 (use -v to see invocation)

besides this it also gives me a warning that a directory has not been found
/Build/Products/Debug-iphonesimulator/React

**Comment by DennisSnijder** [comment]
Never mind, got this fixed by linking it manually

**Comment by rjain-pivotal** [comment]
How did you link manually, I am facing the same error. @DennisSnijder

**Comment by rjain15** [comment]
@DennisSnijder  : How did you link manually, I am facing the same error.

**Comment by DennisSnijder** [comment]
@rjain15 
https://github.com/airbnb/react-native-maps/blob/master/docs/installation.md#option-3-manually
try this, also try to use rnpm instead of npm and do the "rnpm link", saves you some work.

**Comment by rjain15** [comment]
Hi Dennis, 

It still didn't work, I started with a fresh project --- like a Hello World and followed the steps. 

` Ld /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/first.app/first normal x86_64
    cd /Users/rjain/code/react-native-samples/first/ios
    export IPHONEOS_DEPLOYMENT_TARGET=8.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.1.sdk -L/Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator -F/Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator -F/Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/React -F/Users/rjain/code/react-native-samples/first/ios/Pods/GoogleMaps/Subspecs/Base/Frameworks -F/Users/rjain/code/react-native-samples/first/ios/Pods/GoogleMaps/Subspecs/Maps/Frameworks -filelist /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Intermediates/first.build/Debug-iphonesimulator/first.build/Objects-normal/x86_64/first.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -mios-simulator-version-min=8.0 -Xlinker -object_path_lto -Xlinker /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Intermediates/first.build/Debug-iphonesimulator/first.build/Objects-normal/x86_64/first_lto.o -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -ObjC -lc++ -lz -framework Accelerate -framework CoreData -framework CoreGraphics -framework CoreLocation -framework CoreText -framework GLKit -framework GoogleMaps -framework GoogleMapsBase -framework GoogleMapsCore -framework ImageIO -framework OpenGLES -framework QuartzCore -framework React -framework SystemConfiguration -framework UIKit -ObjC -lc++ /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libReact.a /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libRCTActionSheet.a /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libRCTGeolocation.a /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libRCTImage.a /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libRCTLinking.a /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libRCTNetwork.a /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libRCTSettings.a /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libRCTText.a /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libRCTVibration.a /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/libRCTWebSocket.a -lAirMaps -framework Pods_first -Xlinker -dependency_info -Xlinker /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Intermediates/first.build/Debug-iphonesimulator/first.build/Objects-normal/x86_64/first_dependency_info.dat -o /Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/first.app/first
ld: warning: directory not found for option '-F/Users/rjain/Library/Developer/Xcode/DerivedData/first-giisolpkkzrrqedkjvilwcfpgzqi/Build/Products/Debug-iphonesimulator/React'
ld: framework not found React
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

**Comment by Vinu-MysteriousTechVision** [comment]
I have the same issue in using pod. Manually linking is not working for google maps

**Comment by vinceprofeta** [comment]
@Vinu-MysteriousTechVision did you find a solution?

**Comment by qq326646683** [comment]
@rjain15 I have the same error,have you deal with it?

**Comment by Vinu-MysteriousTechVision** [comment]
@vinceprofeta solved the issue for me .
Given bellow procedure that i followed to solve the issue.
1. create fresh project
2. install react-native-maps (and **importend that i haven't link the project**)
3. **Option 2: CocoaPods [https://github.com/airbnb/react-native-maps/blob/master/docs/installation.md]( https://github.com/airbnb/react-native-maps/blob/master/docs/installation.md)** pod install procedure used to link the project. this option is used for my project related. 

**note**: if you getting the same error also do one think that clear the derived data, clean the project and build

**Comment by evansiroky** [comment]
I had a project where I had installed via CocoaPods where it worked on the iOS emulator, but then got this error when trying to build for the device.  I tried making a test project and everything seemed to work fine after running the command `react-native link react-native-maps` in the test project.  So I then proceeded to remove CocoaPods from my existing xcode project using [this method](http://stackoverflow.com/questions/16427421/how-to-remove-cocoapods-from-a-project/34765245#34765245).  I also deleted the reference to AirMaps.xcodeproj from Libraries in the Xcode IDE and then re-ran `react-native link react-native-maps`.  I then setup the signing and everything seemed to work after that.
